creusot/tests/should_fail/final_borrows.rs

```diff
@@ -11,18 +11,20 @@ pub fn not_final_borrow<T: Copy>(bor: &mut T) {
 
 pub fn store_changes_prophecy<T: Copy>(bor: &mut T) {
     let b1 = &mut *bor;
+    let b1_snap = snapshot!(b1);
     let val = *b1;
     // The prophecy of `bor` changed here after the reborrow
     *bor = val;
-    proof_assert!(b1 == bor);
+    proof_assert!(*b1_snap == bor);
 }
 
 pub fn store_changes_prophecy_2<T: Copy>(bor: &mut T, x: T) {
     let b1 = &mut *bor;
     *b1 = x;
+    let b1_snap = snapshot!(b1);
     // The prophecy of `bor` changed here after the reborrow
     *bor = x;
-    proof_assert!(b1 == bor);
+    proof_assert!(*b1_snap == bor);
```

Comment:
*That* is weird. I guess you added the snap in order to compensate the fact that `b1` gets uninitialized here. But What is the instruction that does so?

Reply:
Ok, now I tested and I see that removing the snap does not change anything. Why did you add it?

Reply:
Does it ?? I think I got confused then